analyzers/src/SonarAnalyzer.CSharp/Rules/Utilities/TokenTypeAnalyzer.cs

```diff
@@ -128,7 +128,8 @@ context.Parent switch
                 {
                     // some identifier can be bound to a type or a constant:
                     CaseSwitchLabelSyntax => ClassifyIdentifierByModel(name), // case i:
-                    var x when ConstantPatternSyntaxWrapper.IsInstance(x) => ClassifyIdentifierByModel(name), // is i
+                    BinaryExpressionSyntax { RawKind: (int)SyntaxKind.IsExpression, Right: { } x } when x == context => ClassifyIdentifierByModel(name), // is i
```

Comment:
Please extract the condition to a method. It will make the code easier to understand.